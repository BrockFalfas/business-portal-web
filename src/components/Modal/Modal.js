import React from 'react';
import PropTypes from 'prop-types';
import { Button } from 'antd';

import './Modal.css';

export class Modal extends React.Component {
  static propTypes = {
    children: PropTypes.oneOfType([PropTypes.node, PropTypes.func]),
    onClose: PropTypes.func,
    title: PropTypes.string,
  };

  static defaultProps = {
    className: 'slideInUp',
    onClose: () => {},
  };

  render() {
    const { children, className, onClose, title } = this.props;
    return (
      <div className={`Modal animated ${className}`}>
        <div className="Modal__body">
          <div className="Modal__container">
            <div className="Modal__header">
              <Button
                className="Modal__header--close-button"
                type="default"
                shape="circle"
                icon="close"
                size="default"
                onClick={onClose}
              />
              <div className="Modal__header--title">{title}</div>
            </div>
            <div className="Modal__content">{children}</div>
          </div>
        </div>
      </div>
    );
  }
}

export default Modal;
