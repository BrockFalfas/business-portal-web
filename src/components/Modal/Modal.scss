@import "../../styles/var";

.Modal {
  background-color: #fff;
  z-index: 999;
  position: absolute;
  width: 100%;
  top: 0;
  left: 0;
  height: 100vh;
  overflow-x: hidden;
  &--hidden {
    display: none;
  }
  &__body {
    height: auto;
    background-color: #fff;
  }
  &__container {
    width: 900px;
    margin: 0 auto;
    height: 100%;

    @media (min-width: $screen-xs-min) {
      width: $screen-xs;
    }
    @media (min-width: $screen-sm-min) {
      padding: 1rem 3rem;
      width: $screen-sm;
    }
    @media (min-width: $screen-md-min) {
      width: $screen-md;
    }
    @media (min-width: $screen-lg-min) {
      width: $screen-lg;
    }
    @media (min-width: $screen-xl-min) {
      width: 900px;
    }
  }
  &__header {
    margin-top: 3rem;
    &--title {
      display: inline-block;
      font-size: 1.5rem;
      font-weight: 600;
      padding-left: 2rem;
      vertical-align: middle;
    }
    &--close-button {
      display: inline-block;
      margin-left: 0.5rem;
    }
  }
  &__content {
    padding: 3rem 0;
  }
}
