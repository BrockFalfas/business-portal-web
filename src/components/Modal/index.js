import React from 'react';
import PropTypes from 'prop-types';
import { Button } from 'antd';

import classNames from 'classnames';

import './Modal.css';

export class Modal extends React.Component {
  static propTypes = {
    children: PropTypes.oneOfType([PropTypes.node, PropTypes.func]),
    title: PropTypes.string,
  };

  constructor(props) {
    super(props);
    this.state = {
      isVisible: false
    };

    this.close = this.close.bind(this);
  }

  close() {
    this.setState({
      isVisible: false
    })
  }

  open(event) {
    this.setState({
      isVisible: true
    })
  }

  render() {
    const { children, title } = this.props;
    const { isVisible } = this.state;
    return (
      isVisible ?
        <div className={classNames('Modal animated', { slideInUp: isVisible })}>
          <div className="Modal__body">
            <div className="Modal__container">
              <div className="Modal__header">
                <Button className="Modal__header--close-button" type="default" shape="circle" icon="close"
                        size="default"
                        onClick={this.close}/>
                <div className="Modal__header--title">
                  {title}
                </div>
              </div>
              <div className="Modal__content">
                {children}
              </div>
            </div>
          </div>
        </div>
        : <div/>
    );
  }
}

export default Modal;
