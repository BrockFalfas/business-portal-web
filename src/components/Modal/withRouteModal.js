import React from 'react';

import Modal from './Modal';

const withRouteModal = ({ component: Component, ...modalProps }) => {
  const HOC = props => {
    const closeCallback = () => props.history.goBack();

    return (
      <Modal onClose={closeCallback} {...modalProps}>
        <Component onClose={closeCallback} {...props} />
      </Modal>
    );
  };
  const displayName = Component.displayName || Component.name || '';
  HOC.displayName = `withRouteModal(${displayName}`;
  return HOC;
};

export default withRouteModal;
