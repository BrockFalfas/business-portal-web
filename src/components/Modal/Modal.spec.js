import React from 'react';
import { shallow } from 'enzyme';

import { Modal } from './Modal';

const initModal = overrides => {
  const mockProps = {
    onClose: jest.fn(),
  };
  const wrapper = shallow(<Modal {...mockProps} {...overrides} />);
  return { mockProps, wrapper };
};

describe('component: <RefreshButton/>', () => {
  describe('RefreshButton', () => {
    it('should render without crashing', () => {
      const { wrapper } = initModal();
      expect(wrapper).toMatchSnapshot();
    });
    it('should find button', () => {
      const { wrapper } = initModal();
      expect(wrapper.find('Button.Modal__header--close-button')).toBeTruthy();
    });
    it('should check if onClose function works', () => {
      const { wrapper, mockProps } = initModal();
      wrapper.find('Button.Modal__header--close-button').simulate('click');
      expect(mockProps.onClose).toHaveBeenCalled();
    });
  });
});
