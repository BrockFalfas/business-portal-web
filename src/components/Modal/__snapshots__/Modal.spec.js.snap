// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`component: <RefreshButton/> RefreshButton should render without crashing 1`] = `
<div
  className="Modal animated slideInUp"
>
  <div
    className="Modal__body"
  >
    <div
      className="Modal__container"
    >
      <div
        className="Modal__header"
      >
        <Button
          className="Modal__header--close-button"
          icon="close"
          onClick={[Function]}
          shape="circle"
          size="default"
          type="default"
        />
        <div
          className="Modal__header--title"
        />
      </div>
      <div
        className="Modal__content"
      />
    </div>
  </div>
</div>
`;
